packages/ember-glimmer/lib/utils/lookup-component.js

```diff
@@ -7,14 +8,40 @@ function lookupComponentPair(componentLookup, owner, name, options) {
   let component = componentLookup.componentFor(name, owner, options);
   let layout = componentLookup.layoutFor(name, owner, options);
 
-  if (!layout && component) {
-    let layoutProp = component.proto().layout;
-    if (layoutProp) {
-      let templateFullName = 'template:components/' + name;
-      owner.register(templateFullName, layoutProp);
+  if (component) {
+    let templateFullName = 'template:components/' + name;
+    let {
+      templateForName,
```

Comment:
I think we should emit a different opcode or the dynamic opcode when a layout is late bound and assert on the layout property like the tag expression is asserted.